include/effectmanager.h

```diff
@@ -507,6 +509,83 @@ class EffectManager : public IJSONSerializable
         SaveEffectManagerConfig();
     }
 
+    // Creates a copy of an existing effect in the list. Note that the effect is created but not yet added to the effect list;
+    //   use the AppendEffect() function for that.
+    std::shared_ptr<LEDStripEffect> CopyEffect(size_t index)
+    {
+        if (index >= _vEffects.size())
+        {
+            debugW("Invalid index for CopyEffect");
+            return nullptr;
+        }
+
+        static size_t jsonBufferSize = JSON_BUFFER_BASE_SIZE;
+
+        auto& sourceEffect = _vEffects[index];
+
+        std::unique_ptr<AllocatedJsonDocument> ptrJsonDoc = nullptr;
+
+        SerializeWithBufferSize(ptrJsonDoc, jsonBufferSize,
+            [&sourceEffect](JsonObject &jsonObject) { return sourceEffect->SerializeToJSON(jsonObject); });
+
+        auto jsonEffectFactories = g_EffectFactories.GetJSONFactories();
+        auto factoryEntry = jsonEffectFactories.find(sourceEffect->EffectNumber());
+
+        if (factoryEntry == jsonEffectFactories.end())
+            return nullptr;
+
+        auto copiedEffect = factoryEntry->second(ptrJsonDoc->as<JsonObjectConst>());
+
+        if (!copiedEffect)
+            return nullptr;
+
+        copiedEffect->SetEnabled(false);
+
+        return copiedEffect;
+    }
+
+    // Adds an effect to the effect list and enables it. If an effect is added that is already in the effect list then the result
+    //   is undefined but potentially messy.
+    bool AppendEffect(std::shared_ptr<LEDStripEffect>& effect)
+    {
+        if (!effect->Init(_gfx))
+            return false;
+
+        _vEffects.push_back(effect);
+        EnableEffect(_vEffects.size() - 1, true);
+
+        SaveEffectManagerConfig();
+
+        return true;
+    }
+
+    bool DeleteEffect(size_t index)
+    {
+        if (index >= _vEffects.size())
+        {
+            debugW("Invalid index for DeleteEffect");
+            return false;
+        }
+
+        // We don't allow core effects to be deleted
+        if (_vEffects[index]->IsCoreEffect())
+            return false;
+
+        if (index == _iCurrentEffect)
+            NextEffect();
+
+        DisableEffect(index, true);
+
+        _vEffects.erase(_vEffects.begin() + index);

```

Comment:
Fancy, for some reason I've never done this before!

Reply:
Neither had I. :)